### DISCUSSION TITLE
[CRUD Writable Views] It's not clear how to configure the pipeline for a one-to-many relation (multilookup)

### DISCUSSION THREAD

**Discussion by alexBonadei** [mia-platform/community#328] [Q&A]
### Which section?

Marketplace

### Which is the version?

.

### Description

Hi, I'm having troubles when setting up a writable view.

I have the `orders` collection where there is a field called `dishes` which is an array of strings representing the ids of the ordered dishes:
```
{
_id: ObjectId('.....'),
dishes: ['652552e4fe1a7dade996a730', '652552e4fe1a7dade996a731']
}
```

I wrote a pipeline for the view that correctly returns this:
```
{
_id: ObjectId('.....'),
"dishes": [
        {
          "value":"652552e4fe1a7dade996a730",
          "name": "dish1"
        },
        {
          "value":"652552e4fe1a7dade996a731",
          "name": "dish2"
        }
    ]
}
```

However the POST does not work as I expect. 

The docs states this:
`Have at least one $lookup step in the view pipeline followed by $unwind to make sure only one object it is returned;`

But after the lookup step I don't need to unwind, because then I'll be forced to group them back.
Probably the pipeline I wrote is not compatible.

My pipeline performs these steps:
1. $addFields: It basically converts the strings in the dishes array in objectIds for performing the lookup
2. $lookup: performs the lookup
3. $project

Am I missing something? Thank you in advance

**Comment by danibix95** [comment]
Hi @alexBonadei! We acknowledged the unclear documentation regarding Writable Views and the fact that multi-lookup was not properly working.

Consequently, we [implemented](https://github.com/mia-platform/crud-service/pull/223) a fix for configuring multi-lookups in aggregation pipeline and reviewed the documentation. It is available in the new CRUD Service version, `v6.9.3`.  
However, we recommend to upgrade to version `v6.9.4`, which also includes a startup improvement when indexes should be created.
